### DISCUSSION TITLE
#105 problems with radial-gradients?

### DISCUSSION THREAD

**Discussion by verberg** [cssbattle/forum#243] [Q&A] [answered]
<img width="695" alt="Bildschirmfoto 2022-03-25 um 11 54 39" src="https://user-images.githubusercontent.com/94834403/160108381-9d79b56e-33a5-4ee2-9cd0-f48fb7b4a369.png">

My solution doesn't get 100% match, and on https://codepen.io/alexzaworski/full/vwmNKr I see that the black ring is not sharp as expected. Can someone explain me why? Shall I look for another way to solve this target?

**Comment by gunnarbittersmann** [comment]
Gradients are intended to be used for – ehm gradients, i.e. gradual color changes. Color steps aren’t their intended use-case. Hence browsers don’t apply anti-aliasing when rendering gradients. And you can’t even say that they’re doing it wrong.
So yeah, if you can’t make a gradient work, try something else.

**Reply by verberg** [reply]
But sometimes - even often! -  it works fine and I really wonder what the reason is.

**Reply by gunnarbittersmann** [reply]
Anti-aliasing and [display resolution](https://github.com/cssbattle/forum/discussions/2#discussioncomment-2437148).

**Reply by rafehimself** [reply]
@verberg have you tried pixels ? or just q,
sometimes 10px works but 10q or 11q not

**Reply by verberg** [reply]
Yes I have, even fractions of pixels !

**Reply by rafehimself** [reply]
then replace those gradients with shadows and check if that will give you 100%,
cuz that might not be the problem after all, usually they worked better than border-radius, unless there has been so update in scoring

**Reply by verberg** [reply]
I get 100% match now width box-shadows!

**Comment by SelenIT** [comment, marked as answer]
My solution uses radial gradients and gets 100% despite ragged edges. Both `21q` and `5vw` work.
![image](https://user-images.githubusercontent.com/282754/160143084-c23ff5a7-6d84-4e71-b46f-a221fb9f4bb4.png)

Maybe the problem is in the position of the "shadows" or the rotation angles of the eyes?

**Reply by verberg** [reply]
Thanks a lot. I've tried again and got 100% match and actually I am on range 11 (!)

**Reply by verberg** [reply]
@SelenIT "despite ragged edges" I habe this problem now with #103 (super saiyan) and can't get 100%. I tried different ways and have no idea how to solve it. Any tip?

**Reply by SelenIT** [reply]
I have to admit that I didn't even try radial gradients for [#103](https://cssbattle.dev/play/103) 😳 I used the method that is now exposed thanks to [#106](https://cssbattle.dev/play/106) 😊 (and still it took me a lot of attempts to get 100%)

**Reply by midnightPush** [reply]
@SelenIT how to generate that path?

**Reply by SelenIT** [reply]
I have used to write them by hand, fragment by fragment, starting from the general direction and adjusting values by trial and error until the shape fits (more or less). It's not that hard after some practice

**Reply by Gabb192** [reply]
@SelenIT And where can i practice? I can't figure out where to put commas but also the meaning "q, v, m, a, h etc..", is there a site or a guide?

**Reply by gunnarbittersmann** [reply]
The [search for ~Spo~ path in MDN](https://developer.mozilla.org/en-US/search?q=path&w=3&qs=plugin) might be a good start. Who would‘ve thought?

**Reply by Gabb192** [reply]
I'll check it out.

**Reply by verberg** [reply]
#103 has anyone solved this target without a path? I'v tried so many ways, I give up now!

**Reply by Gabb192** [reply]
@verberg Yes me, but mine is 600+ characters 😢

**Reply by midnightPush** [reply]
no, it's very broken target,
does admin not checking it before they release it❓  🙄 🤷 🙈 😕 

how do you do vertical hair without clip-path?

**Reply by Gabb192** [reply]
@midnightPush Oh no wait you right, for te vertical hair I used clip-path

**Reply by verberg** [reply]
@Gabb192 I solved the "the vertical hair" without clip-path, and this part of my code seems to be perfect, but I really don't know, why the others doesn't much 100%. I put one of the codes below (a more complicate one)!
`<p><p><p><p i><p i><style>*{background:linear-gradient(var(--b,#161616 100%,0,#161616))}*>*{--b:90deg,#FFF,50%,#FFDEB9 0;margin:var(--q,160 160 84);border-radius:var(--r,0 0 90q 90q)}p{position:fixed;width:60;aspect-ratio:1;--b:195deg,#0000 50%,0,#EBAE11;--r:50%;--q:-29.8-19.7;-webkit-box-reflect:right var(--c,-1.5px);z-index:-1}p+p{;--b:240deg,#0000 50%,0,#EBAE11;--q:-48.1-5.4;--c:-29.7px;z-index:1}p+p+p{width:120;--b:90deg,#EBAE11 17.3px,0,#0000;--q:-94 22.6;--c:-205px}[i]{width:60;height:50;--b:#0000 50%,0,#EBAE11;--q:-41 10;--c:n}[i]+p{width:20;height:20;--b:#0000 50%,0,#161616;--q:26 30`

**Reply by Gabb192** [reply]
@verberg I tried a lot too, but not seems to work.. Btw what's your name on the site?

**Reply by verberg** [reply]
@Gabb192 VerenaBerg -- and yours?

**Reply by Gabb192** [reply]
@verberg Gabriele. Do you have discord?

**Reply by verberg** [reply]
@gabb192 Sorry, my English is not so good, what do you want me to ask?

**Reply by Gabb192** [reply]
@verberg Nothing, just talk about the solution.

**Comment by Gabb192** [comment]
Same problem here (99.9%):

![aa](https://user-images.githubusercontent.com/101881425/160281693-907cc560-10b1-4267-8b1b-b1fe3525c3a5.png)


But in this case with box-shadow:

`box-shadow:inset 8q 13px#5A6042,inset 0 0 0 32q#868A64,inset 0 0 0 37q;`

**Comment by gunnarbittersmann** [comment]
I took me a while to get my first 100% match on this target. My problem was `*+*{-webkit-box-reflect:…}` which affected not only the `body` but also other elements – not visible to my eyes or a diff tool, but the server does her own thing. With `body{-webkit-box-reflect:…}`, it matched.

**Reply by joewinbpaul** [reply]
> I took me a while to get my first 100% match on this target. My problem was `*+*{-webkit-box-reflect:…}` which affected not only the `body` but also other elements – not visible to my eyes or a diff tool, but the server does her own thing. With `body{-webkit-box-reflect:…}`, it matched.

"her"?? u mean the server is a female?!!

**Reply by NaviTalib** [reply]
can you give me full soluion?
